### PR TITLE
Teapot example: rotation added

### PR DESCRIPTION
**Pull request by Batanick**
Had troubles with figuring out how to actually update buffers after creation (see #254). So I've added a simple rotation to the teapot example, maybe would be useful to someone

### CODE REVIEW

File: examples/src/bin/teapot_vs.glsl [hunk 1]
```diff
@@ -18,11 +18,13 @@ layout(location = 1) in vec3 normal;
...
+    mat4 view;
     mat4 proj;
 } uniforms;
 
 void main() {
-    v_normal = transpose(inverse(mat3(uniforms.worldview))) * normal;
-    gl_Position = uniforms.proj * uniforms.worldview * vec4(position, 1.0);
+    mat3 worldview = mat3(uniforms.view * uniforms.world);
+    v_normal = transpose(inverse(worldview)) * normal;
+    gl_Position = uniforms.proj * uniforms.view * uniforms.world * vec4(position, 1.0);
```
Comment (tomaka) [inline examples/src/bin/teapot_vs.glsl]: Please replace `uniforms.view * uniforms.world` with `worldview` since you're calculating it already.

**Comment by tomaka** [comment]
Thanks!

r+ after the minor fix.

**Comment by Batanick** [comment]
fixed

**Comment by nicokoch** [comment]
Hey, I noticed the teapot does not rotate when I launch the executable directly like `./target/debug/teapot`.
When invoking with `cargo run --bin teapot` everything works fine.

Is this supposed to happen?

**Comment by tomaka** [comment]
> Hey, I noticed the teapot does not rotate when I launch the executable directly like ./target/debug/teapot.
When invoking with cargo run --bin teapot everything works fine.

`examples/target/debug/teapot` probably refers to an old compilation.
Since vulkano uses Cargo workspaces, the *real* `target` directory is now at the root of the repo.

**Comment by nicokoch** [comment]
Yep that's it, thanks. I wish `cargo clean` would delete the old .target dirs

**Pull request merged**